mllib/src/test/scala/org/apache/spark/ml/regression/GeneralizedLinearRegressionSuite.scala

```diff
@@ -83,10 +83,11 @@ class GeneralizedLinearRegressionSuite
       testData.toDF()
     }
 
+    // force some labels to be exactly zero
     datasetPoissonLog = generateGeneralizedLinearRegressionInput(
-      intercept = 0.25, coefficients = Array(0.22, 0.06), xMean = Array(2.9, 10.5),
+      intercept = -1.5, coefficients = Array(0.22, 0.06), xMean = Array(2.9, 10.5),
       xVariance = Array(0.7, 1.2), nPoints = 10000, seed, noiseLevel = 0.01,
-      family = "poisson", link = "log").toDF()
+      family = "poisson", link = "log").map{x => LabeledPoint(if (x.label < 0.7) 0.0 else x.label, x.features)}.toDF()
```

Comment:
I'd suggest creating a test case explicitly to check that Poisson accepts 0 values. Then we can just use some very small dataset (5 points or similar) which contains a zero label. 

The way it is now, if someone were to change the intercept to some other large number in the future, then we wouldn't have any zero labels in the data and we wouldn't exercise that test case. 

Also, you can run the style checker via `./dev/lint-scala` to check that the patch meets style guidelines. This line is too long, currently.